crosshair/stubs_parser.py

```diff
@@ -104,28 +126,36 @@ def _sig_from_ast(
                 except Exception:
                     debug("Not able to evaluate condition:", assign_text)
                 if condition is not None:
-                    sig = _sig_from_ast(
+                    new_sigs, valid = _sig_from_ast(
                         node.body if condition else node.orelse,
                         next_steps,
                         stub_text,
                         glo,
                     )
-                    if sig:
-                        return sig
+                    sigs.extend(new_sigs)
+                    is_valid = is_valid and valid
 
-    return None
+    return sigs, is_valid
 
 
 def _exec_import(
     imp: Union[ast.Import, ast.ImportFrom], stub_text: str, glo: Dict[str, Any]
 ):
     """Try to execute the import statement and add it to the `glo` namespace."""
-    # Ignore imports from `_typeshed`
     for n in imp.names:
         if n.name == "_typeshed":
             return
+    # Replace imports from `_typeshed` by their equivalent
     module = getattr(imp, "module", None)
     if module == "_typeshed":
+        for n in imp.names:
+            name = n.name
+            if name in _REPLACE_TYPESHED:
+                new_module, replace = _REPLACE_TYPESHED[name]
+                exec("from " + new_module + " import " + replace + " as " + name, glo)
```

Comment:
Consider [importlib.import_module](https://docs.python.org/3/library/importlib.html#importlib.import_module)  (+getattr) for this.

Reply:
I agree this would be better this way. I'll update it.